### DISCUSSION TITLE
Welcome to BetterViewer!

### DISCUSSION THREAD

**Discussion by Ademking** [Ademking/BetterViewer#15] [General]
# 👋 Welcome to BetterViewer

I'm starting an experiment using Discussions as a place to talk about BetterViewer. I hope that you:
- Ask questions you're wondering about.
- Share ideas.
- Get help with a PR you're working on

To get started, comment below with an introduction of yourself and tell us about what you do with BetterViewer!

**Comment by Ademking** [comment]
## 👋 Hello and Welcome to BetterViewer!

BetterViewer makes image viewing faster, easier, and more fun. ❤️
It was designed as a replacement for the image viewing mode built into Firefox and Chrome-based web browsers.
With BetterViewer you can use various keyboard shortcuts to quickly pan, zoom images, edit and a lot more!
GitHub: https://github.com/Ademking/BetterViewer

I'd love to hear your thoughts and feel free to drop in any questions!

**Comment by sabbirahmed2005** [comment]
Thank you so much

**Reply by Ademking** [reply]
You're welcome @sabbirahmed2005
Thank you so much for your feedback.

**Comment by W0ofles** [comment]
INSTANTLY added this to my favorite add-ons/extensions list. Possibly the most underrated extension I have EVER seen!!!.......For now(; 
This is going to blow up, man. Thank you SO much for making this and releasing it OPEN SOURCE! Says a lot about you as a developer. 
I am happy to support you and your future projects!

**Reply by Ademking** [reply]
Hi @W0ofles,
Thank you so much for your positive feedback and taking time out of your day to leave me this super kind review. You made my day! I'm so happy to hear that you are enjoying BetterViewer.

**Comment by hthanh100** [comment]
good

**Reply by Ademking** [reply]
Thank you @hthanh100 ♥

**Comment by MOHAMED-EHAB-DEV** [comment]
Very Amazing

**Reply by Ademking** [reply]
Thank you @Programming-School-Pro-Coding ♥

**Comment by stugib76** [comment]
This is brilliant! Just what I was looking for.

It'd be great to have an option to send the resulting manipulated image back to the clipboard, either through a button/keyboard shortcut, or even automatically, e.g. I crop an image and then the cropped image can be pasted into another app.

**Reply by Ademking** [reply]
Thank you @stugib76 so much for your positive feedback and taking time out of your day to leave me this kind review.

You can crop / edit your image and copy the resulting image. Just Right-click to copy to the clipboard.

![image](https://user-images.githubusercontent.com/12462188/175016280-a5284593-bdc0-4184-9138-d0d1ace04130.png)

**Reply by Squirrelllll** [reply]
brilliant! no suggestions yet

**Comment by mohammed199924** [comment]
This is probably the most polished chrome extension I've ever used, thanks for the outstanding work!
I would like suggest two main features that would make this laterally perfect:
1- fill to screen (vertically, horizontally)
2- auto hide tools, appears when mouse is at screen corner/right or left edges, (having that tool bar be vertical would make this perfect).

Also, how can I buy you guys coffee?

**Reply by dragon1020** [reply]
I agree, extension is about perfect. I miss one more (with fill and auto hide) thing - "Show all images on this page" menu, like in "Image Viewer" extension (no longer available in the Store).

**Comment by Jackpot118** [comment]
great sis, the photo editor is very professional, cheers sis harley myanmar

**Comment by Jackpot118** [comment]
great sis, the photo editor is very professional, cheers sis harley Myanmar

**Comment by Jordy3D** [comment]
I'd been using a different extension for changing the default image viewer for a while, but it starting failing more and more often. 
The second I tried this out, I was happy I did.

**Comment by Seroko111** [comment]
THIS.IS.INCREDIBLE.
Thanks!

**Comment by namename-123** [comment]
this doesn't apply to any image I open anymore (https://*.*/*.jpg,png,jpeg etc) 
please revert this change NOW, or at least add an option to do so, I really liked this as an automatic replacement for the default viewer I don't want to manually open each image from now on.......

**Reply by Ademking** [reply]
@namename-123 This change is due to Google Manifest V3 which restricts certain features. Unfortunately, it’s not possible to auto-open images in BetterViewer anymore, as it operates under the same mechanism. 
You can still right-click on an image and select "Open Image with BetterViewer" as before.

![image](https://github.com/user-attachments/assets/dfe4b03e-8d48-4cdc-906d-6f610c89a45d)

**Reply by namename-123** [reply]
I'm using Brave, which allows manifest v2 extensions to work. Can I revert to the older version and prevent it from updating? Thank you for your time!

**Reply by Ademking** [reply]
> I'm using Brave, which allows manifest v2 extensions to work. Can I revert to the older version and prevent it from updating? Thank you for your time!

Sure thing 👍 

1/ Download this version from here:
https://github.com/Ademking/BetterViewer/releases/download/1.0.5/betterviewer-chrome.zip

2/ Unzip it to a folder. 7zip should do the trick. Put it somewhere where you won't delete it, of course.

3/ Go to brave://extensions

4/ Enable 'Developer mode'. The toggle is on the top-right of the tab.

5/ Press 'Load unpacked' and select the unzipped folder.

Done

**Reply by namename-123** [reply]
I did just that, thank you. Ramadan Mubarak and good luck on your project!

**Reply by ViperGeek** [reply]
Thanks for the tip on downgrading to 1.0.5.  Are there any other enhancements, improvements, and/or bug fixes in 2.0.1 that we'd be missing by manually installing 1.0.5?

**Reply by Ademking** [reply]
@ViperGeek There isn't a big difference in the core functionalities ... You can try both versions

**Comment by ayianadarko** [comment]
Stopped working in Brave

**Reply by Ademking** [reply]
It works fine. Try to use latest version

![image](https://github.com/user-attachments/assets/9b1e1ebb-d42f-4a1a-a250-00d0c6b3281c)

**Comment by ViperGeek** [comment]
Is there a way to bind a keyboard shortcut to Open Image in BetterViewer?  I tried using Brave's ability to define shortcuts, but "Activate Extension" isn't quite doing it (technically, it seems to do nothing).

If there is no current way, would you be open to a feature request to add the ability?

**Comment by justfetus** [comment]
Incredible extension, thank you for providing it.

I'm relatively new to using it so may be wondering about features that existed pre-Manifest V3...
Are we still expecting a "customize toolbar" feature, or am I missing it somewhere?
What about the change background? (I don't want the image blurred in the background)
And finally - any way to edit keyboard shortcuts / would that be possible? Ctrl + D for saving an image is a bit unintuitive to me.

**Comment by ViperGeek** [comment]
**Disclaimer:** If I should be posting these enhancement requests as Issues in GitHub, please let me know.

Would it be possible to allow the user to adjust the zoom rate when using the mouse wheel?  Currently, the zoom jumps from 100% to 150% to 225% — 1.5x per step — which is usually a bit too quick. Being able to cut this down to 1.2x would be perfect.

**Comment by ViperGeek** [comment]
I just manually upgraded to BetterViewer 2.0.2 (2.0.1 is still “current” on the Chrome store) and when I try to right-click and open an image in a new tab, I get the following error:
```
ehmdoalmdmafbfghhdaaeedgenajmgel is blocked
This page has been blocked by Brave
ERR_BLOCKED_BY_CLIENT
```
<img width="572" alt="BetterViewer-2 0 2" src="https://github.com/user-attachments/assets/66517353-b233-4304-a36e-cba44f00239c" />


Is there something I need to enable in the extension to allow Brave, and possibly other browsers, to access the image content?

**Reply by ViperGeek** [reply]
**UPDATE:** This only happens in a Private window in Brave. False alarm!

Note: BetterViewer doesn't take control of "Open image in new tab" if the extension Tab Activate is also enabled.

**Comment by MouayadYT** [comment]
I love this extension! I was wondering, is there a way to make this page not appear when my computer restarts and i open my browser?

**Reply by temp10min** [reply]
Same here, it opens every time.

**Reply by NetMage** [reply]
This is annoying enough I think I am going to turn BetterViewer off until it is fixed.

**Comment by Reaaaaa** [comment]
so good, thank you so much for developing this!

**Comment by Sebi11927** [comment]
I like better viewer, but this "welcome to BetterViewer" message opens pointlessly many times 😠

**Reply by Ademking** [reply]
@Sebi11927  The issue has been fixed in version 2.0.3. You can try it right away by downloading it here:
https://github.com/Ademking/BetterViewer/releases/tag/2.0.3
Or you can wait for the update to roll out automatically via the Chrome Web Store.

**Reply by temp10min** [reply]
Could you update it on the Edge store as well, please?

**Comment by rayleighZ210** [comment]
Thanks for the great extension. 
Quick question, on Firefox Open Image in BetterViewer works, but Open Image in New Tab doesn't open the image with BetterViewer. Is this the expected behavior in Firefox? 
Tested on Firefox 141.0 (ver 2.02)

**Comment by AlconburyBlues** [comment]
Apparently the extension just broke, as of today.  3-Aug-25.  I use it on 2 different browsers.  Waterfox and Firefox.  And it stopped working on both of them at the same time.  So I'm guessing that's not a browser issue.  Because Waterfox and Firefox are managed by 2 different groups of devs.  It's highly unlikely they both released a bad patch for both browsers in the same day.  I tried removing the extension, restarting each browser, and readding the extension.  Still doesn't work.  I'm guessing the extension itself got a breaking update.  Been using this for years, and it's great.  So I hope it gets fixed.

**Reply by riotinprogress** [reply]
I was wondering if I had screwed up settings or a different extension had messed everything up, but nope, I'm having the same issue. BetterViewer is no longer working when opening an image in a new tab. It does work when "Open with BetterViewer" is used.

I'm using the latest release of Zen Browser.

**Reply by C0smicCactus** [reply]
Just found this extension, just installed it and instantly thought "yeah, it's neat but it would be better if it replaced to default viewer". Bummer about the bug but it's nice to know that is the intended design. Hope it gets fixed soon. Also on Zen Browser.

**Comment by Reaaaaa** [comment]
same here :/

**Comment by spider623** [comment]
surprised of how functional it is, if possible please consider adding JXL support as it's back on Chrome Canary for now, and soon in stable without the need of a flag
